src/test/java/net/bitnine/agensgraph/test/jdbc/AgArrayTest.java

```diff
@@ -0,0 +1,230 @@
+            assertEquals("vertex", v[i].getType());
+        }
+
+        rs.close();
+        stmt.close();
+    }
+
+    @Test
+    public void testVertexResultSet() throws SQLException {
+        Statement stmt = conn.createStatement();
+        ResultSet rs = stmt.executeQuery("MATCH p=()-[]->()-[]->() RETURN nodes(p)");
+        assertTrue(rs.next());
+        Array arr = rs.getArray(1);
+        rs = arr.getResultSet();
+
+        assertTrue(rs.next());
+        assertEquals(1, rs.getInt(1));
+        Vertex v1 = (Vertex) rs.getObject(2);
+        assertEquals(1, v1.getInt("vid"));
+
+        assertTrue(rs.next());
+        assertEquals(2, rs.getInt(1));
+        Vertex v2 = (Vertex) rs.getObject(2);
+        assertEquals(2, v2.getInt("vid"));
+
+        assertTrue(rs.next());
+        assertEquals(3, rs.getInt(1));
+        Vertex v3 = (Vertex) rs.getObject(2);
+        assertEquals(3, v3.getInt("vid"));
+
+        assertTrue(!rs.next());
+
+        rs.close();
+        stmt.close();
+    }
+
+    @Test
+    public void testGraphIdArray() throws SQLException {
+        Statement stmt = conn.createStatement();
+        ResultSet rs = stmt.executeQuery("MATCH (v)-[]->() RETURN array_agg(id(v))");
+        assertTrue(rs.next());
+        Array arr = rs.getArray(1);
+        GraphId[] gid = (GraphId[]) arr.getArray();
+
+        for (int i = 0; i < gid.length; i++) {
+            assertEquals("graphid", gid[i].getType());
+        }
+
+        rs.close();
+        stmt.close();
+    }
+
+    @Test
+    public void testGraphIdResultSet() throws SQLException {
+        Statement stmt = conn.createStatement();
+        ResultSet rs = stmt.executeQuery("MATCH ()-[r]->() RETURN array_agg(id(r))");
+        assertTrue(rs.next());
+        Array arr = rs.getArray(1);
+        rs = arr.getResultSet();
+
+        assertTrue(rs.next());
+        assertEquals(1, rs.getInt(1));
+        GraphId gid1 = (GraphId) rs.getObject(2);
+        assertEquals("graphid", gid1.getType());
+
+        assertTrue(rs.next());
+        assertEquals(2, rs.getInt(1));
+        GraphId gid2 = (GraphId) rs.getObject(2);
+        assertEquals("graphid", gid2.getType());
+
+        assertTrue(!rs.next());
+
+        rs.close();
+        stmt.close();
+    }
+
+    @Test
+    public void testGetBaseTypeName() throws SQLException {
+        Statement stmt = conn.createStatement();
+        ResultSet rs = stmt.executeQuery("MATCH p=()-[]->()-[]->() RETURN relationships(p)");
+        assertTrue(rs.next());
+        Array arr = rs.getArray(1);
+
+        assertEquals("edge", arr.getBaseTypeName());
+
+        rs.close();
+        stmt.close();
+    }
+
+    @Test
+    public void testGetBaseType() throws SQLException {
+        Statement stmt = conn.createStatement();
+        ResultSet rs = stmt.executeQuery("MATCH p=()-[]->()-[]->() RETURN relationships(p)");
+        assertTrue(rs.next());
+        Array arr = rs.getArray(1);
+
+        assertEquals(2002, arr.getBaseType());
```

Comment:
There is `Types.STRUCT` for `2002` here.

Reply:
It looks good to use Types.STRUCT.